tests/mpi/broadcast_01.cc

```diff
@@ -80,4 +115,11 @@ main(int argc, char *argv[])
 
   // broadcast test
   check({-1, 0, 1}, "i love democracy");
+
+  // broadcast test
+  check(23);
+
+  // broadcast test
+  double d = 14.32;
+  check(d);
```

Comment:
Let's also check some more exotic types here, such as `std::complex<float>`.

Reply:
Fixed in 7573ec9